src/main/java/org/sonatype/nexus/ci/util/Nxrm3Util.groovy

```diff
@@ -32,18 +34,29 @@ class Nxrm3Util
       throw new IllegalArgumentException('Specified Nexus Repository Manager instance is not a 3.x server')
     }
 
-    getApplicableRepositories(configuration.serverUrl, configuration.credentialsId)
+    getApplicableRepositories(configuration.serverUrl, configuration.credentialsId, 'maven')
   }
 
   /**
-   * Return Nexus repositories which are applicable for package upload. These are maven2 hosted repositories.
+   * Return Nexus repositories which are applicable for package upload. These are hosted repositories matching
+   * the provided format.
    */
-  static List<Repository> getApplicableRepositories(final String serverUrl, final String credentialsId) {
-    nexus3Client(serverUrl, credentialsId).getRepositories().
-        findAll { 'hosted'.equalsIgnoreCase(it.type) && 'maven2'.equalsIgnoreCase(it.format) }
+  static List<Repository> getApplicableRepositories(final String serverUrl, final String credentialsId,
+                                                    @Nullable final String format) {
+    nexus3Client(serverUrl, credentialsId).getRepositories()
+        .findAll { ('hosted'.equalsIgnoreCase(it.type)) && (format != null ? format.equalsIgnoreCase(it.format) : true)}
```

Comment:
Probably best to also check for empty string here: `NexusStringUtils.isNotBlank(format) ? ...`

Reply:
Good call

Reply:
trivial: since this is groovy could also consider making format have a default value of null, which could simplify line 60 (and any future usage) below to not even pass the null as an argument.

Reply:
@wrightkris totally forgot about that option in groovy